### ISSUE TITLE
Dilation of keras.Conv2DTranspose does not do anything

### ISSUE THREAD

**Issue by wagenrace** [tensorflow/tensorflow#21598]
### Describe the problem
Well using tf.keras.Conv2DTranspose I noticed the dilation function did not dilate. No matter what size of dilation you choose the result is the same.

I did found the problem in keras as well but I did not now if I should report it here too: https://github.com/keras-team/keras/issues/8159

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes, see below
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 x64
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**: N/A
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.8.0
- **Python version**: 3.6.4
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: v8.0
- **GPU model and memory**: GeForce 1050
- **Exact command to reproduce**: See below

### Source code / logs
```
import tensorflow as tf
import numpy as np

Conv2D = tf.keras.layers.Conv2D
Conv2DTranspose = tf.keras.layers.Conv2DTranspose
Input = tf.keras.layers.Input
Model = tf.keras.Model

input = Input(shape=(None, None, 1), name='input')
x = Conv2DTranspose(8, (2, 2), dilation_rate=2, padding='valid')(input)
x = Conv2DTranspose(8, (3, 3), dilation_rate=2, padding='valid')(x)
x = Conv2DTranspose(8, (3, 3), dilation_rate=2, padding='valid')(x)

model_dilation = Model(inputs=input, outputs=x, name='dil')

x = Conv2DTranspose(8, (2, 2), dilation_rate=1, padding='valid')(input)
x = Conv2DTranspose(8, (3, 3), dilation_rate=1, padding='valid')(x)
x = Conv2DTranspose(8, (3, 3), dilation_rate=1, padding='valid')(x)

model_no_dilation = Model(inputs=input, outputs=x, name='no_dil')

output_dil = model_dilation.predict(np.zeros((1, 5, 5, 1)))
output_no_dil = model_no_dilation.predict(np.zeros((1, 5, 5, 1)))

print("shape dil", np.shape(output_dil))
print("shape no dil", np.shape(output_no_dil))
```

result:
```
shape dil (1, 10, 10, 8)
shape no dil (1, 10, 10, 8)
```

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
Bazel version
Exact command to reproduce
Mobile device

**Comment by wagenrace** [comment]
Good bot
Everything is updated

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yanboliang** [comment]
@wagenrace @robieta I have sent #22265 to fix this, would you mind to take a look? Thanks.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @robieta: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
